### ISSUE TITLE
React warning: "Can't perform a React state update on a component that hasn't mounted yet"

### ISSUE THREAD

**Issue by alex-statsig** [relay-tools/relay-hooks#233]
When using React 18 with Suspense, I get a warning "Can't perform a React state update on a component that hasn't mounted yet. This indicates that you have a side-effect in your render function that asynchronously later calls tries to update the component. Move this work to useEffect instead.".

I'm guessing the query is finishing while the component is suspended, and this library is triggering state updates when it's not really mounted. It is nice that the library starts querying right away before its technically mounted, but should refrain from actually updating state I think

**Comment by morrys** [comment]
Can you create a minimal sample project that reproduces the error so I can investigate?

**Comment by gino** [comment]
Having the same issues here unfortunately. 

Using Vite (4.1.0), with vite-plugin-relay (2.0.0) with React (18.2.0) and relay-hooks (7.2.0).


<img width="1488" alt="CleanShot 2023-03-02 at 13 14 15@2x" src="https://user-images.githubusercontent.com/23482089/222425908-3b7cc240-67be-4cd6-b1c9-f949b754f135.png">

**Comment by morrys** [comment]
Hi @gino,
I have many tests that try the render problems usingLazyLoadQuery and none detect these problems.
Not even my example project https://github.com/relay-tools/relay-hooks/tree/master/examples/suspense/nextjs-ssr which I also just upgraded to the latest versions of react, nextjs and relay-hooks.

To better understand the problem I need a minimal example project where I can check it out

**Comment by alex-statsig** [comment]
I'm trying to set up a repro but it seems maybe not as simple as I expected. I had thought the cause was straightforward: a component suspending while the useQuery hook has a request in flight. I'll try and get something together soon.

**Comment by alex-statsig** [comment]
Ah looks like I've done it, I'm not sure why one of the changes was necessary. Also seemed to uncover a (likely related) side-effect of the issue, that the component will repeatedly re-render/re-query while suspended.

https://github.com/alex-statsig/relay-hooks-bug-repro

**Comment by morrys** [comment]
the only way is to allow the update only after the components have mounted.

The solution is also simple. Just change the forceUpdate:
```js
import { useCallback, useEffect, useReducer, useRef } from 'react';
export function useForceUpdate() {
  var _a = useReducer(function (x) {
    return x + 1;
  }, 0),
      forceUpdate = _a[1];

  var isMounted = useRef(false);
  useEffect(function () {
    isMounted.current = true;
    return function () {
      isMounted.current = false;
    };
  }, []);
  var update = useCallback(function () {
    if (isMounted.current) {
      forceUpdate();
    }
  }, [isMounted, forceUpdate]);
  return update;
}
```

The tests run successfully and your project works fine too, but at the moment I have doubts that there may be other cases missing that I need to check

**Comment by morrys** [comment]
fixed in version 8.0.0 https://github.com/relay-tools/relay-hooks/releases/tag/v8.0.0
